### PR TITLE
Test cat unique count

### PR DESCRIPTION
**Pull request by thayeylolu**
- [ ] Write basic unit test
- [ ] include a sample dataset
- [ ] include parametrisation of bad input
-

### CODE REVIEW

File: tests/test_cat_unique_counts.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        9,
+        df_sample()
+    ]
+)
+def test_cat_unique_counts_bad_input(bad_input):
+    """Accepts parameters and rasie errors when parameter is of wrong instance"""
+    with pytest.raises(TypeError):
+        cat_unique_counts(bad_input)
+
+def test_cat_unique_counts():
```
Comment (Anthea98) [inline tests/test_cat_unique_counts.py:45]: we need 3-5 edge cases~maybe one more case~

File: tests/test_cat_unique_counts.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                                ["Marry", 19, '1'],
+                                ['Clarke', 1, '4']], 
+                           columns=['Name','Age', "Rate"])
+
+    result1 = list(cat_unique_counts(df1).iloc[:, 0])
+    result2 = list(cat_unique_counts(df2).iloc[:, 0])
+
+    assert result1 == ['Name', 'Hobby']
+    assert result2 == ['Name']
+
```
Comment (Anthea98) [inline tests/test_cat_unique_counts.py:62]: we need 3-5 edge cases in the unit test, maybe add one more case~

File: tests/test_cat_unique_counts.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        df_sample()
+    ]
+)
+def test_cat_unique_counts_bad_input(bad_input):
+    """Accepts parameters and rasie errors when parameter is of wrong instance"""
+    with pytest.raises(TypeError):
+        cat_unique_counts(bad_input)
+
+def test_cat_unique_counts():
+    """Unit test for missing value counts."""
```
Comment (Anthea98) [inline tests/test_cat_unique_counts.py:46]: """Unit test for unique value counts."""

File: tests/test_cat_unique_counts.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    df2 = pd.DataFrame(data = [['Alex', 10, '3'],
+                                ["Marry", 19, '1'],
+                                ['Clarke', 1, '4']], 
+                           columns=['Name','Age', "Rate"])
+
+    result1 = list(cat_unique_counts(df1).iloc[:, 0])
+    result2 = list(cat_unique_counts(df2).iloc[:, 0])
+
+    assert result1 == ['Name', 'Hobby']
+    assert result2 == ['Name']
```
Comment (Anthea98) [inline tests/test_cat_unique_counts.py:61]: Looks good! How about adding 1-2 tests related to the count number we got? like how many "Alex" here hhh

**Review by Anthea98 — CHANGES_REQUESTED** [review]
GOOD JOB! some suggestions are here~

File: tests/test_cat_unique_counts.py (hunk 3 above)
Comment (thayeylolu) [reply tests/test_cat_unique_counts.py:46]: this is what happens when you copy and paste someone else's code😆

File: tests/test_cat_unique_counts.py (hunk 4 above)
Comment (thayeylolu) [reply tests/test_cat_unique_counts.py:61]: Great idea.  I will include it. Thank you for the feed back

File: tests/test_cat_unique_counts.py (hunk 2 above)
Comment (thayeylolu) [reply tests/test_cat_unique_counts.py:78]: Thank you Anthea. I will include it

**Review by Anthea98 — APPROVED** [review]
nice work~
